File: libs/document-processor-next/src/registry/processors/operation-marker-processor.ts
import { blueIds as conversationBlueIds } from '@blue-repository/conversation';

import type { OperationMarker } from '../../model/index.js';
import type { MarkerProcessor } from '../types.js';
import { operationMarkerSchema } from '../../model/index.js';

export class OperationMarkerProcessor
  implements MarkerProcessor<OperationMarker>
{
  readonly kind = 'marker' as const;
  readonly blueIds = [conversationBlueIds['Operation']] as const;
  readonly schema = operationMarkerSchema;
}

File: references/java-sdk/src/main/java/blue/language/processor/registry/processors/OperationMarkerProcessor.java
package blue.language.processor.registry.processors;

import blue.language.processor.ContractProcessor;
import blue.language.processor.model.OperationMarker;

public class OperationMarkerProcessor implements ContractProcessor<OperationMarker> {

    @Override
    public Class<OperationMarker> contractType() {
        return OperationMarker.class;
    }
}
